hushline/routes.py

```diff
@@ -228,7 +226,9 @@ def register() -> Response | str:
                 flash("💔 Username already taken.", "error")
                 return redirect(url_for("register"))
 
-            new_user = User(primary_username=username, password=password)
+            # Create new user instance
```

Comment:
this looks like a useful change is being reverted

Reply:
We're just splitting this into a two-step process. The change handles hashing in the `password_hash` setter.

```
new_user = User(primary_username=username)
new_user.password_hash = password
```

model.py:
```
    @password_hash.setter
    def password_hash(self, plaintext_password):
        """Hash plaintext password using scrypt and store it."""
        self._password_hash = scrypt.hash(plaintext_password)  
```